### PR TITLE
IC-1625 remove 'using_rar_days' from referral and rename 'rar' to 'enforceable' elsewhere

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

IC-1625 remove 'using_rar_days' from referral and rename 'rar' to 'enforceable' elsewhere

## What is the intent behind these changes?

make the naming consistent with the UI and the biz terms

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceTest.kt [hunk 1]
```diff
@@ -297,22 +297,6 @@ class ReferralServiceTest @Autowired constructor(
     referralService.updateDraftReferral(sampleReferral, DraftReferralDTO(hasAdditionalResponsibilities = true, whenUnavailable = "wednesdays"))
   }
 
-  @Test
-  fun `when usingRarDays is true, maximumRarDays must be set`() {
```
Comment (sldblog) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceTest.kt:301]: Could we add a test please about what happens if the `maximumEnforceableDays` is null when we try to send the referral?

Comment (tomsmyers) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceTest.kt:301]: sure thing. having said that i'm not sure we explicitly have tests for any other required fields having been set... 

perhaps something to add in another changeset to validate all the required fields?

**Review by sldblog — APPROVED** [review]
👍 Let's specify preconditions to sending a referral in a separate PR
